core/state_processor.go

```diff
@@ -146,6 +163,12 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 	receipt.BlockHash = blockHash
 	receipt.BlockNumber = blockNumber
 	receipt.TransactionIndex = uint(statedb.TxIndex())
+	if len(input) == 36 {
```

Comment:
consider dropping the else statement.
```
if len(input) != 36 {
  panic(fmt.Errorf("unexpected input length: %d", len(input)))
}
[..]
```

Reply:
this is a very early PR, I think most of this was just a debugging harness